src/AndcultureCode.CSharp.Sitefinity.Conductors/Extensions/DynamicContentExtensions.cs

```diff
@@ -0,0 +1,116 @@
+﻿using AndcultureCode.CSharp.Sitefinity.Core.Attributes;
+using AndcultureCode.CSharp.Sitefinity.Core.Models.Content;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using Telerik.Sitefinity.DynamicModules;
+using Telerik.Sitefinity.DynamicModules.Model;
+using Telerik.Sitefinity.GenericContent.Model;
+using Telerik.Sitefinity.Model;
+using Telerik.Sitefinity.RelatedData;
+using Telerik.Sitefinity.Utilities.TypeConverters;
+
+namespace AndcultureCode.CSharp.Sitefinity.Conductors.Extensions
+{
+    public static class DynamicContentExtensions
+    {
+        public static DynamicContent SetDataItemProperties<T>(this T item, PropertyInfo[] properties, string providerName) where T : SitefinityContent
+        {
+            SitefinityMetadataAttribute metadataAttribute = (SitefinityMetadataAttribute)item.GetType().GetCustomAttribute(typeof(SitefinityMetadataAttribute));
+
+            if (metadataAttribute == null)
+            {
+                throw new Exception($"Could not cast attribute to {nameof(SitefinityMetadataAttribute)}");
+            }
+
+            var dynamicContentType = metadataAttribute.DynamicContentType;
+
+            if (dynamicContentType == null)
+            {
+                return null;
+            }
+
+            var resolvedDynamicContentType = TypeResolutionService.ResolveType(dynamicContentType);
+
+            var dynamicModuleManager = DynamicModuleManager.GetManager(providerName);
+            var dataItem = dynamicModuleManager.CreateDataItem(resolvedDynamicContentType);
+
+            properties.ToList().ForEach(property => SetPropertyInDataItem(
+                item,
+                metadataAttribute,
+                ref dataItem,
+                property,
+                providerName
+            ));
+
+            return dataItem;
+        }
+
+        private static void SetPropertyInDataItem<T>(
+           T item,
+           SitefinityMetadataAttribute metadataAttribute,
+           ref DynamicContent dataItem,
+           PropertyInfo property,
+           string providerName
+       )
+        {
+            if (!property.PropertyType.IsSubclassOf(typeof(SitefinityContent)) &&
```

Comment:
Minor: Persist the `IsSubclassOf` results in variables for readability, and so you don't have to write the same function call on line 65.